src/core/weightedGraph.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+import {Graph, type GraphJSON} from "./graph";
+import {type Weights as WeightsT, type WeightsJSON} from "./weights";
+import * as Weights from "./weights";
+import {toCompat, fromCompat, type Compatible} from "../util/compat";
+
+export type WeightedGraph = {|
+  +graph: Graph,
+  +weights: WeightsT,
+|};
```

Comment:
What are the semantics of a `WeightedGraph`?

For instance, is it the case that each address in `weights` corresponds
to a node/edge in the graph (say, that the weights are *valid*)? Is it
also the case that each node/edge in the graph has a weight (say, that
the weights are *complete*)?

Is there any meaning associated with weights, or any natural domain for
them? For instance, for some kinds of weighted graphs, negative and zero
weights make sense, and combination along paths is done additively. For
others, only positive weights make sense, and combination is done
multiplicatively. In others, weights may be any real or ±∞, with
combination given by max(—, —). (The underlying theme here is usually
that [the weights have the structure of a semiring][app].) Does the type
defined in this module have an opinion?

[app]: https://www.cs.colostate.edu/~cs575dl/Sp2015/Lectures/APP.pdf

Reply:
The type defined in this module is unopinionated with respect to all of these questions. All it does is provide a way to package weights alongside a graph. I've added a docstring to this effect.

The Weights type itself is unopinionated about how weight composition works. The semantics are defined in weightEvaluator.js:

https://github.com/sourcecred/sourcecred/blob/9784927402bf4fde8716949b71092d6dcd42d72d/src/analysis/weightEvaluator.js#L16-L28

However, it's likely that we will create more evaluators in the future, which apply different interpretations to the weights.